wolfSSL/src/chapter02.md

```diff
@@ -394,301 +394,1177 @@ Removes code for OAEP padding.
+
+#### WOLFSSL_APPLE_NATIVE_CERT_VERIFICATION
+
+Enables the use of Apple's native trust APIs when authenticating TLS peer certificates. Requires [WOLFSSL_SYS_CA_CERTS](#WOLFSSL_SYS_CA_CERTS) to be defined. This macro does not need to be set by the user if building with `configure` or `CMake` on iOS or other apple devices, but should be explicitly set on MacOS if you wish to use the native verification methods.
+
+#### WOLFSSL_CUSTOM_CURVES
+
+Allow non-standard curves. Includes the curve "a" variable in calculation. Additional curve types can be enabled using [`HAVE_ECC_SECPR2`](#have_ecc_secpr2), [`HAVE_ECC_SECPR3`](#have_ecc_secpr3), [`HAVE_ECC_BRAINPOOL`](#have_ecc_brainpool) and [`HAVE_ECC_KOBLITZ`](#have_ecc_koblitz).
+
+#### HAVE_COMP_KEY
+
+Enables ECC compressed key support.
+
+#### WOLFSSL_EXTRA_ALERTS
+
+Enables additional alerts to be sent during a TLS connection. This feature is also enabled automatically when [`--enable-opensslextra`](#--enable-opensslextra) is used.
+
+#### WOLFSSL_DEBUG_TLS
+
+Enables additional debugging print outs during a TLS connection
+
+#### HAVE_BLAKE2
+
+Enables Blake2s algorithm support
+
+#### HAVE_FALLBACK_SCSV
+
+Enables Signaling Cipher Suite Value(SCSV) support on the server side. This handles the cipher suite 0x56 0x00 sent from a client to signal that no downgrade of TLS version should be allowed.
+
+#### HAVE_AEAD
+
+Implements the use of AEAD and is required for TLS 1.3.
+
+#### HAVE_AES_CBC
+
+Used In the case when Static RSA is needed.
+
+#### HAVE_AES_DECRYPT
+
+Enables AES to decrypt symmetric cypher.
+
+#### HAVE_ALPN
+
+Crypto enables the option for application-layer protocol negotiation.
+
+#### HAVE_CAVIUM_OCTEON_SYNC
+
+By default, the OCTEON's global variables are all thread local. This tag allows them to be shared between threads.
+
+#### HAVE_CERTIFICATE_STATUS_REQUEST
+
+Used for Certificate revocation as a cert status request feature.
+
+#### HAVE_CERTIFICATE_STATUS_REQUEST_V2
+
+Used for Certificate revocation as a cert status request feature.
+
+#### HAVE_IO_TIMEOUT
+
+Certificate revocation. IO options enable support for connect timeout, but the default is off.
+
+#### HAVE_CURL
+
+Used for building a subset of the wolfSSL library when linking with cURL.
+
+#### HAVE_CURVE448
+
+Define for Curve448 support. Additional macro settings can be changed. The default is to enable shared secret, key export, and import.
+
+#### HAVE_DANE
+
+To verify Raw Public Key cert, DANE(RFC6698) add macro HAVE_DANE for dane authentification.
```

Comment:
"authentication", not "authentification".

Btw DANE is an interesting protocol that uses DNSSEC (domain lookup with crypto signatures) to allow certificates to be validated without depending on the usual certificate authority stuff.  It actually provides an enormously stronger guarantee of legitimacy.